idf_build_apps/app.py

```diff
@@ -302,6 +338,7 @@ def _process_sdkconfig_files(self):
         if not os.path.isdir(expanded_dir):
             os.makedirs(expanded_dir)
 
+        self._process_override_sdkconfig_file()
         for f in self._sdkconfig_defaults + ([self.sdkconfig_path] if self.sdkconfig_path else []):
```

Comment:
if you create one file here, based on the `--override-sdkconfig` and `--override-sdkconfig-file`, for example

```suggestion
        _override_sdkconfig_temp_file = self._process_override_sdkconfig_file() # return the list of the tempfile path: os.path.join(self.work_dir, 'override_sdkconfig_tempfile') or an empty list
        for f in self._sdkconfig_defaults + ([self.sdkconfig_path] if self.sdkconfig_path else [] + _override_sdkconfig_temp_file):
```

all the later lines and other functions could be reused. and the `self._sdkconfig_files` could include the override sdkconfig file you created.

Reply:
I believe there might be some misunderstandings about what I did in the `_process_override_sdkconfig_file` function. What I did was to update the `override_sdkconfig` attribute dictionary with fields from the `override_sdkconfig_file` which putted in app project folder.
To clarify, the overridden file has not been created yet; it will read the actual file from ```for f in self._sdkconfig_defaults + ([self.sdkconfig_path] if self.sdkconfig_path else []):```, and then override its content, just as it did before, and add it to 'res'.